core/dive.c

```diff
@@ -761,7 +778,8 @@ void per_cylinder_mean_depth(const struct dive *dive, struct divecomputer *dc, i
 	uint32_t lasttime = 0;
 	int lastdepth = 0;
 	int idx = 0;
-	unsigned int used_mask, known_mask;
+	bool *used_cylinders;
+	int num_used_cylinders;
```

Comment:
I'm not entirely convinced about the whole "use bool array instead of bits". It's kind of sad. We really don't care about more than 32 cylinders.

But hey, I guess I don't care that deeply.

Reply:
On that specific note... the world record 330m dive that took 15 hours was done with 9 tanks. Even if you want to plan and have the real dive in the same dive, 32bits allows for 16 tanks... I just don't think this is something that's worth changing.

Reply:
On the other hand - who knows what strange use-case the planner comes up with. If we remove the hard-limit, let's remove it completely. Otherwise we'd have to add sanity checks and then how do we treat the failure case, etc.

Reply:
now THAT is the winning argument. Adding all the checks for numCylinders < whatever limit is just silly.